### PR TITLE
Enable customized header text for WAM prompts and web mode in Broker

### PR DESCRIPTION
**Pull request by shalinikhare27**
This PR incorporates changes Rick did for adding customized header text in WAM prompts and web mode.
PR reference - https://github.com/AzureAD/microsoft-authentication-cli/pull/11

Since this PR is already reviewed and has all the details, please check it out if you are missing any context.

### CODE REVIEW

File: src/MSALWrapper.Test/AuthFlowTest/AuthFlowBrokerTest.cs [hunk 1]
```diff
@@ -396,6 +398,27 @@ public async Task BrokerAuthFlow_GetTokenInteractive_OperationCanceledException_
...
+        /// Ensure <see cref="IPCAWrapper.WithPromptHint"/> be invoked in <see cref="TokenFetcherPublicClient.GetTokenNormalFlowAsync"/>.
+        /// </summary>
+        /// <returns>The <see cref="Task"/>.</returns>
+        [Test]
+        public async Task GetTokenNormalFlowAsync_GetTokenInteractive_WithPromptHint()
+        {
+            this.SilentAuthUIRequired();
+            this.InteractiveAuthResult();
+
+            this.MockAccount();
```
Comment (reillysiemens) [inline src/MSALWrapper.Test/AuthFlowTest/AuthFlowBrokerTest.cs:411]: This part isn't present in #11, @shalinikhare27. Can you shed some light on why this is necessary?

File: src/MSALWrapper.Test/AuthFlowTest/AuthFlowBrokerTest.cs [hunk 2]
```diff
@@ -396,6 +398,27 @@ public async Task BrokerAuthFlow_GetTokenInteractive_OperationCanceledException_
             authFlowBroker.ErrorsList[2].Message.Should().Be("Interactive Auth (with extra claims) timed out after 15 minutes.");
         }
 
+        /// <summary>
+        /// Ensure <see cref="IPCAWrapper.WithPromptHint"/> be invoked in <see cref="TokenFetcherPublicClient.GetTokenNormalFlowAsync"/>.
```
Comment (reillysiemens) [inline src/MSALWrapper.Test/AuthFlowTest/AuthFlowBrokerTest.cs:402]: Are these references still valid? 🤔

File: src/MSALWrapper.Test/AuthFlowTest/AuthFlowBrokerTest.cs [hunk 3]
```diff
@@ -396,6 +398,27 @@ public async Task BrokerAuthFlow_GetTokenInteractive_OperationCanceledException_
             authFlowBroker.ErrorsList[2].Message.Should().Be("Interactive Auth (with extra claims) timed out after 15 minutes.");
         }
 
+        /// <summary>
+        /// Ensure <see cref="IPCAWrapper.WithPromptHint"/> be invoked in <see cref="TokenFetcherPublicClient.GetTokenNormalFlowAsync"/>.
+        /// </summary>
+        /// <returns>The <see cref="Task"/>.</returns>
+        [Test]
+        public async Task GetTokenNormalFlowAsync_GetTokenInteractive_WithPromptHint()
```
Comment (reillysiemens) [inline src/MSALWrapper.Test/AuthFlowTest/AuthFlowBrokerTest.cs:406]: Is this test method name still appropriate? The method under test here is no longer called `GetTokenNormalFlowAsync`. I feel like we can probably arrive at a more accurate, terser name here.

**Review by reillysiemens — CHANGES_REQUESTED** [review]
Just a couple small suggested changes first. Otherwise, I think this looks good.

File: src/MSALWrapper.Test/AuthFlowTest/AuthFlowBrokerTest.cs (hunk 2 above)
Comment (shalinikhare27) [reply src/MSALWrapper.Test/AuthFlowTest/AuthFlowBrokerTest.cs]: This is a splendid catch. Fixed it!

File: src/MSALWrapper.Test/AuthFlowTest/AuthFlowBrokerTest.cs (hunk 3 above)
Comment (shalinikhare27) [reply src/MSALWrapper.Test/AuthFlowTest/AuthFlowBrokerTest.cs]: Yep, renamed it to be consistent with other tests in AuthFlowBroker.

File: src/MSALWrapper.Test/AuthFlowTest/AuthFlowBrokerTest.cs (hunk 1 above)
Comment (shalinikhare27) [reply src/MSALWrapper.Test/AuthFlowTest/AuthFlowBrokerTest.cs:411]: I need to MockAccount to mock the method `TryToGetCachedAccountAsync` to get account. You can see the `MockAccount` implementation here and every test in the file uses this method as well. :)

File: src/MSALWrapper.Test/AuthFlowTest/AuthFlowBrokerTest.cs (hunk 2 above)
Comment (reillysiemens) [reply src/MSALWrapper.Test/AuthFlowTest/AuthFlowBrokerTest.cs]: @shalinikhare27: Isn't `IPCAWrapper` also invalid? I thought we did away with that interface... 🤔

Comment (reillysiemens) [reply src/MSALWrapper.Test/AuthFlowTest/AuthFlowBrokerTest.cs]: Oh, nvm, it appears to still be there.

**Review by reillysiemens — APPROVED** [review]

**Review by varshini-elayappen — APPROVED** [review]

**Pull request merged**